progenitor-client/Cargo.toml

```diff
@@ -10,7 +10,10 @@ description = "An OpenAPI client generator - client support"
 bytes = "1.2.1"
 futures-core = "0.3.24"
 percent-encoding = "2.1"
-reqwest = { version = "0.11", default-features = false, features = ["json", "stream"] }
+reqwest = { git = "https://github.com/seanmonstar/reqwest", rev = "6ceb23958c8b6d7f1d8ee093f0ad73184d133d40", features = ["json", "stream"] }
+#reqwest = { version = "0.11", default-features = false, features = ["json", "stream"] }
 serde = "1.0"
 serde_json = "1.0"
 serde_urlencoded = "0.7.1"
+base64 = "0.13"
+rand = "0.8"
```

Comment:
I think these would need to be added here as well: https://github.com/oxidecomputer/progenitor/blob/main/progenitor-impl/src/lib.rs#L426

Is there a reasonable way to avoid these dependencies? "No" is a fine answer...

Reply:
i guess the way to avoid it would be re-implementing them ourselves, which feels pretty far out of scope.
(for what it's worth, we're pulling these in to avoid pulling in all of tungstenite)
this (and potential future such cases of adding new deps) would also be less of an issue if we were to move to emitting code that uses crate re-exports from progenitor-client